code/modules/mob/living/carbon/carbon.dm

```diff
@@ -1186,7 +1182,27 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 	SSticker.score.score_food_eaten++
 	return TRUE
 
-/mob/living/carbon/proc/selfFeed(obj/item/reagent_containers/food/to_eat, fullness)
+/mob/living/carbon/proc/drink(obj/item/reagent_containers/drinks/to_eat, mob/user)
+	if(user == src)
+		if(!selfDrink(to_eat))
+			return FALSE
+	else if(!forceFed(to_eat, user, nutrition))
+		return FALSE
+
+	if(to_eat.consume_sound)
+		playsound(loc, to_eat.consume_sound, rand(10, 50), TRUE)
+	if(to_eat.reagents.total_volume)
+		taste(to_eat.reagents)
+		var/fraction = min(1 / to_eat.reagents.total_volume, 1)
+		var/drink_size = to_eat.amount_per_transfer_from_this < 5 ? 5 : to_eat.amount_per_transfer_from_this
```

Comment:
I don't fully understand reagent code but something about this seems off. If you have a glass with 5u of something, and a transfer amount of 5, the result should be the same as if you have a glass of 5u of something and a transfer amount of 50, but drink_size is dependent on the transfer amount regardless of the amount in the glass.